### ISSUE TITLE
[Tracking] Make Emojis WhatsAppQuality

### ISSUE THREAD

**Issue by stitesExpensify** [Expensify/App#14163]
Old (unreviewed) [doc with mock-ups](https://docs.google.com/document/d/1ctN-C-NBU3eIho9kgl8gzAN3zX_dSK2WyPqXoa52xW0/edit?pli=1#bookmark=id.g0c5zax90l40)

### Outstanding WAQ
- https://github.com/Expensify/App/issues/12578

### Downstream polish

- https://github.com/Expensify/App/issues/12189
- https://github.com/Expensify/App/issues/12263
- https://github.com/Expensify/App/issues/4114
- https://github.com/Expensify/App/issues/14180

### Done
- https://github.com/Expensify/App/issues/12188
- https://github.com/Expensify/App/issues/12691
- https://github.com/Expensify/Expensify/issues/239100
- https://github.com/Expensify/App/issues/8459
- https://github.com/Expensify/App/issues/12804
- https://github.com/Expensify/App/issues/13070
- https://github.com/Expensify/App/issues/13160
- https://github.com/Expensify/App/issues/12772
- https://github.com/Expensify/Expensify/issues/239099
- https://github.com/Expensify/App/issues/13818
- https://github.com/Expensify/App/issues/14302
- https://github.com/Expensify/App/issues/15108
- https://github.com/Expensify/App/issues/14098

**Comment by JmillsExpensify** [comment]
At this point we're officially done with all our unheld WAQ issues, so awesome work so far! The next step we're embarking on it tackling the polish items listed above – starting with #14098 and #12188.

**Comment by JmillsExpensify** [comment]
Both of those will be worked on by Margelo. Then I believe that @stitesExpensify is taking on #239100, which add an emoji category picker.

**Comment by stitesExpensify** [comment]
This is moving along! Margelo has assigned the 2 smaller issues, and are actively working on the design doc for Reactions!

Also, I created this [GitHub Project](https://github.com/orgs/Expensify/projects/21/views/1) (internal only unfortunately) for myself for visual tracking if it helps anybody else. I will keep the project status updated both there and in this issue.

**Comment by stitesExpensify** [comment]
Still moving forward! The detailed design doc is in progress for Emoji Reactions, and I asked for an update on auto-complete here https://github.com/Expensify/App/issues/12188#issuecomment-1408812294

**Comment by stitesExpensify** [comment]
Design doc for Emoji Reactions has released the detailed implementation and is being reviewed
Auto-complete and mobile search are both in review

**Comment by JmillsExpensify** [comment]
Tons of progress happening in this issue. The category picker hit staging (maybe even production by now). A PR is in place for single emojis being larger, same for emoji code auto-suggestions. At that point, we'll be ready to pick up the remaining items on this list, though there are admittedly not many.

**Comment by JmillsExpensify** [comment]
Largely the same as above. Lots of progress in the linked issues in the downstream polish section. Check those for the specific details.

**Comment by stitesExpensify** [comment]
Mobile Search - Actively being worked on and reviewed in PR
Emoji Autosuggestion - Has an approval, but 1 small bug was found
Stickers - I have not started this yet, not a priority
Efficiently store and search for emojis - Needs a design doc, not a priority
Make emojis larger when alone - In review, currently lots of conflicts
Add tooltip for category picker - Merged!
Add emoji reactions - Final review, should be merged first thing tomorrow morning to get it out for ExpensiconX

**Comment by conorpendergrast** [comment]
Hey! In the announcement email today, I see this is a known issue: 
> The hover bar of emojis has the wrong icon (no plus button)

That was also reported in [this issue](https://github.com/Expensify/App/issues/15812), and I can't see a reference to it here. Is there an existing bug report for it? If there isn't, I'll just use #15812 for that! Thanks!

**Comment by stitesExpensify** [comment]
Yep! I made my own issue for that [here](https://github.com/Expensify/App/issues/15787) and already have a PR up :)

**Comment by conorpendergrast** [comment]
Thanks for confirming!

**Comment by JmillsExpensify** [comment]
Emoji autosuggestions have hit staging.

**Comment by stitesExpensify** [comment]
I'm considering making the [sticker issue](https://github.com/Expensify/App/issues/12263) external, or potentially having an expert contributor take it on since it is technically a new feature and not a bug

**Comment by JmillsExpensify** [comment]
@shawnborton What's you'r stance on this one at this juncture? I tend to think we should approach this with a fresh set of eyes and even put this through p/s format. What do you think?

**Comment by JmillsExpensify** [comment]
Building on that last comment, I think the last time we discussed this, we landed on stickers being an incredibly common way to communicate. There's like some insane number of users that send/receive stickers daily, I vaguely remember it being in the hundreds of millions. But whatever the case, I think before we do anything, it'd be good to level set on how we're approaching this.

**Comment by shawnborton** [comment]
I like that idea. I think I still consider this as being different from uploading custom emojis like Slack has. This is more like a sticker pack that's used as a way to express yourself via our new branding and brand universe. I think the UI can be similar to what we've proposed though, or we could even consider a separate sticker icon that's not part of the emoji selector. If we do that, we might want to consider a different compose box layout to support more icons, like these: 
<img width="752" alt="image" src="https://user-images.githubusercontent.com/2319350/232779239-316ac960-e7a3-410d-9920-747f9f74844e.png">


Super rough but you get the idea.

**Comment by JmillsExpensify** [comment]
Awesome yeah that sounds great. I agree with this in particular. 

> I think I still consider this as being different from uploading custom emojis like Slack has.

**Comment by stitesExpensify** [comment]
Yep agreed. I think that as long as we provide the assets for stickers, it can be all front end and we can have a contributor (or expert contributor) take care of it since it should just be the same as adding attachments to a chat!

**Comment by shawnborton** [comment]
I still think I agree with Jason in that we should have a P/S for this, and we should consider a few things for a pre-design as there are a few ways we could implement this. For example, we could have a dedicated sticker button in the chat composer like I shared above (which adds another icon to the composer and thus might warrant moving things around), or we could just supplement the current emoji menu with a sticker section. 

Either way, I don't think I personally have time to run this through the P/S process but I'm happy to help out with mocks if anyone feels passionate.

**Comment by JmillsExpensify** [comment]
Also happy to help but don't have any extra bandwidth.

**Comment by JmillsExpensify** [comment]
Still working through the items listed in the OP

**Comment by JmillsExpensify** [comment]
Not exactly a top priority right now though see individual issues for the latest.

**Comment by stitesExpensify** [comment]
Our focus was switched to threads for a while, but I am coming back to this next week.

**Comment by stitesExpensify** [comment]
The only WAQ issue still open is on hold for an issue on hold for another issue so that will take a while. The rest is just polish

**Comment by stitesExpensify** [comment]
Same update as last week.

**Comment by JmillsExpensify** [comment]
One issue knocked off the list. The emoji reactions issue shouldn't be far behind.

**Comment by JmillsExpensify** [comment]
I think we should close this one out as soon as the issue for emoji reactions closes.

**Comment by stitesExpensify** [comment]
Agreed!

**Comment by stitesExpensify** [comment]
We are very very close! There are a couple of regressions that popped up from the big migration to the new way of emoji reacting, and then we can close this out!

**Comment by stitesExpensify** [comment]
The final regressions are still being taken care of here

**Comment by JmillsExpensify** [comment]
Agreed, let's close it out.
